### PR TITLE
Prepare to merge null safety

### PR DESCRIPTION
**Pull request by natebosch**
- Switch to a major version bump.
- Append a `-dev` to the version since we aren't publishing yet.
- Add `publish_to: none` for safety.
- Reformat `.travis.yml`. Indent list elements to the same position as
  the containing key. This matches yaml style in other Google files.
- Update SDK constraints to use the `2.11.0` dev SDKs.
- Renames some variables with noise word "the" to match the field name
  and avoid the shadowing with `this.`.

### CODE REVIEW

File: pubspec.yaml [hunk 1]
```diff
@@ -1,16 +1,21 @@
...
+ sdk: '>=2.11.0-0 <2.11.0'
 
 dependencies:
   collection: ^1.15.0-nullsafety.2
 
 dev_dependencies:
   test: ^1.16.0-nullsafety.1
+
+dependency_overrides:
+  analyzer: any
```
Comment (kevmoo) [inline pubspec.yaml:21]: Why is this needed? Would love an explanation...

**Review by kevmoo — APPROVED** [review]

File: pubspec.yaml (hunk 1 above)
Comment (natebosch) [reply pubspec.yaml:21]: Analzyer depends on this package, so the breaking version change causes a conflict until we update the dependency.

https://github.com/dart-lang/sdk/blob/d63d5d93a59b370bf5f5c277109110cb67a96b75/pkg/analyzer/pubspec.yaml#L21
